code/mapping/ext_modules/mapping_common/map.py

```
@@ -535,32 +532,63 @@ def is_lane_free(
+            return lane_state, None
 
-            # use is_lane_free_rectangle function as fallback
-            return self.is_lane_free_rectangle(
-                right_lane=right_lane,
-                lane_length=lane_length,
-                lane_transform=lane_transform,
-                reduce_lane=reduce_lane,
-                min_coverage_percent=min_coverage_percent,
-                min_coverage_area=min_coverage_area,
-            )
+        if lane_box is None or not lane_state == LaneFreeState.TO_BE_CHECKED:
```

Ruff (0.8.2)

Use `lane_state != LaneFreeState.TO_BE_CHECKED` instead of `not lane_state == LaneFreeState.TO_BE_CHECKED`

Replace with `!=` operator

(SIM201)